azurerm/internal/services/databasemigration/resource_arm_database_migration_service.go

```diff
@@ -0,0 +1,227 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmDatabaseMigrationService() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmDatabaseMigrationServiceCreate,
+		Read:   resourceArmDatabaseMigrationServiceRead,
+		Update: resourceArmDatabaseMigrationServiceUpdate,
+		Delete: resourceArmDatabaseMigrationServiceDelete,
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.DatabaseMigrationServiceID(id)
+			return err
+		}),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateName,
+			},
+
+			"location": azure.SchemaLocation(),
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"subnet_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"sku_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					// No const defined in go sdk, the literal listed below is derived from the response of listskus endpoint.
+					// See: https://docs.microsoft.com/en-us/rest/api/datamigration/resourceskus/listskus
+					"Premium_4vCores",
+					"Standard_1vCores",
+					"Standard_2vCores",
+					"Standard_4vCores",
+				}, false),
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceArmDatabaseMigrationServiceCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DatabaseMigration.ServicesClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	resourceGroup := d.Get("resource_group_name").(string)
+	name := d.Get("name").(string)
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for present of existing Data Migration Service (Service Name %q / Group Name %q): %+v", name, resourceGroup, err)
+			}
+		}
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_data_migration_service", *existing.ID)
+		}
+	}
+
+	skuName := d.Get("sku_name").(string)
+	subnetID := d.Get("subnet_id").(string)
+	location := d.Get("location").(string)
+
+	parameters := datamigration.Service{
+		Location: utils.String(location),
```

Comment:
_minor_ we can just do these inline if you'd like
```suggestion
		Location: utils.String(azure.NormalizeLocation(d.Get("location").(string))),
```